tawazi/node/node.py

```diff
@@ -229,12 +237,34 @@ def execute(self, results: Dict[Identifier, Any], profiles: Dict[Identifier, Pro
         with profiles[self.id]:
             # 2 post-
             # 2.1 write the result
-            results[self.id] = self.exec_function(*args, **kwargs)
+            try:
+                results[self.id] = self.exec_function(*args, **kwargs)
+            except Exception as e:
+                if self.call_location:
+                    raise TawaziBaseException(
+                        f"Error occurred while executing ExecNode {self.id} at {self.call_location}"
+                    ) from e
+
+                raise e
 
         # 3. useless return value
         logger.debug("Finished executing {} with task {}", self.id, self.exec_function)
         return results[self.id]
 
+    def get_call_location(self) -> str:
+        """Get Location where ExecNode was called."""
+        frame = inspect.currentframe()
```

Comment:
doesn't this kind of manipulation do overhead ?

Reply:
yes it does.
1. This overhead only occurs during build time
2. I optimized the overhead as much as possible. I don't use inspect.stack() which is extremely slow, but I use inspect.currentframe() and then go back 2 frames to get the call location, and this is extremely fast! it is unnoticeable during the DAG build

Reply:
ok works for me then